business/core/home/stores/homedb/homedb.go

```diff
@@ -0,0 +1,216 @@
+package homedb
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/ardanlabs/service/business/core/home"
+	db "github.com/ardanlabs/service/business/data/dbsql/pq"
+	"github.com/ardanlabs/service/business/data/order"
+	"github.com/ardanlabs/service/business/data/transaction"
+	"github.com/ardanlabs/service/foundation/logger"
+	"github.com/google/uuid"
+	"github.com/jmoiron/sqlx"
+)
+
+// Store manages the set of APIs for home database access.
+type Store struct {
+	log *logger.Logger
+	db  sqlx.ExtContext
+}
+
+// NewStore constructs the api for data access.
+func NewStore(log *logger.Logger, db *sqlx.DB) *Store {
+	return &Store{
+		log: log,
+		db:  db,
+	}
+}
+
+// ExecuteUnderTransaction constructs a new Store value replacing the sqlx DB
+// value with a sqlx DB value that is currently inside a transaction.
+func (s *Store) ExecuteUnderTransaction(tx transaction.Transaction) (home.Storer, error) {
+	ec, err := db.GetExtContext(tx)
+	if err != nil {
+		return nil, err
+	}
+
+	s = &Store{
+		log: s.log,
+		db:  ec,
+	}
+
+	return s, nil
+}
+
+// Create adds a Home to the database. It returns an error if something went wrong
+func (s *Store) Create(ctx context.Context, hme home.Home) error {
+	const q = `
+      INSERT INTO homes
```

Comment:
Check you tabbing in the editor.

Reply:
What kind of tabbing do you use? I have it on 2 spaces right now. Ooh the string might be not taken into account.

Reply:
Use 4 spaces per tab. That is the Go idiom.  That is not the problem here. I think you have some combination of tab and space with these queries.